src/core/types/api/index.ts

```diff
@@ -142,4 +146,10 @@ export type Query = Record<string, string | number | (string | number)[]>;
  * * generic messages and signals content,
  * * published message metadata.
  */
-export type Payload = string | number | boolean | { [key: string]: Payload | null } | Payload[];
+export type Payload =
```

Comment:
Just re verifying. Does this Payload includes Date related types?

Reply:
`Date` implements `.toJSON()` function, so it will work as well.